Userland/Libraries/LibIPC/Message.cpp

```diff
@@ -75,8 +76,8 @@ ErrorOr<void> MessageBuffer::transfer_message(Core::LocalSocket& socket)
             if (auto error = maybe_nwritten.release_error(); error.is_errno()) {
                 // FIXME: This is a hacky way to at least not crash on large messages
                 // The limit of 100 writes is arbitrary, and there to prevent indefinite spinning on the EventLoop
-                if (error.code() == EAGAIN && writes_done < 100) {
-                    sched_yield();
+                if ((error.code() == EAGAIN || error.code() == EMSGSIZE) && writes_done < 100) {
+                    Core::EventLoop::current().pump();
```

Comment:
If we're sending a sync message, pumping the event loop seems.. unexpected?

Reply:
True... @alimpfard does this come up in any possible sync messages on serenity? As far as I can tell, this will only happen if we fail to send the message. So if it's too large, or we're blocked up on the receiving side, essentially.

Reply:
actually, ipc never yields to the event loop in sync messages to begin with, it explicitly blocks on the socket.

Reply:
see https://github.com/LadybirdBrowser/ladybird/blob/de6d99e94005b5f5b66b1bf9b83f358d75cc3d69/Userland/Libraries/LibIPC/Connection.cpp#L206-L208 and https://github.com/LadybirdBrowser/ladybird/blob/de6d99e94005b5f5b66b1bf9b83f358d75cc3d69/Userland/Libraries/LibIPC/Connection.cpp#L94

Reply:
Right, it seems wrong to violate the expectation that sending a sync message will not execute any other code until the message has been sent and responded to.

Reply:
Perhaps we could use a separate kind of message to ensure this requirement is met where actually needed (but that's unrelated)

I'll make a PR to fully block if we're sending sync messages, though I haven't seen any issues so far in serenity with this patch.